sdlib/d/gc/page.d

```diff
@@ -796,22 +831,30 @@ private:
 					continue;
 				}
 
-				// If the cycle do not match, all the elements are dead.
-				if ((w & 0xff) != gcCycle) {
+				auto slotsWithMeta = e.slabMetadataFlags.rawContent[0];
+
+				// If the cycle do not match, all the elements are dead. If no
+				// metadata exists, then we can safely clear the whole thing.
+				if (slotsWithMeta == 0 && (w & 0xff) != gcCycle) {
 					deadExtents.insert(e);
 					continue;
 				}
 
 				auto oldOccupancy = e.slabData.rawContent[0];
 				auto newOccupancy = oldOccupancy & (w >> 8);
+				auto toRemove = oldOccupancy ^ newOccupancy;
+
+				// Call any finalizers on dying slots.
+				finalizeSlabNimble(slotsWithMeta & toRemove, ec.sizeClass, e,
```

Comment:
If you move `lotsWithMeta & toRemove` out of the call and assign it to a variable. The variable name will make the code easier to understand and the call itself will look better.

Reply:
I did this, but I don't love it. I feel like it's just too many temporary variables. I suppose the optimizer will remove them?